lib/schema/src/openctx.schema.ts

```diff
@@ -1,18 +1,66 @@
 /**
  * Metadata about code
  */
-export type Schema = Item | Annotation
+export type Schema = CandidateItem | Item | Annotation
 /**
  * A hint about how to best present an annotation to the human in the client's user interface.
  *
  * - `prefer-link-over-detail`: Prefer to show the item as a link over showing the detail text, if the client does not cleanly support doing both.
  */
 export type PresentationHint = 'prefer-link-over-detail'
+/**
+ * Additional information which can be used to idenfity the resource by the provider.
+ */
+export type ItemMeta = { [key: string]: any }
 
+export type ItemKindID = string
+
+/**
+ * A kind of context item returnable by the provider.
+ */
+export interface ItemKind<K extends ItemKindID = ItemKindID> {
+    /**
+     * The identifier for the item kind.
+     */
+    id: K
+    /**
+     * The human-readable name of the item kind.
+     */
+    name: string
+    /**
+     * A human-readable description of the item kind.
+     */
+    description?: string
+}
+/**
+ * A candidate item contains contextual information required to list and identify the resource.
+ */
+export interface CandidateItem<K extends ItemKindID = ItemKindID> {
```

Comment:
Do we need to have CandidateItem be a separate type? It would be better for it to be an Item but with more leniency on fields being undefined. That would present a much simpler API surface imo. It would also make it easier to know if a candidate item is already present in the message, and to make other utility funcs to work with it (like to render it).

Reply:
I like this suggestion a lot :+1:

Reply:
Removed the separate type and using a single Item type only.